benchmarks/dynamo/operatorbench/operators/__init__.py

```diff
@@ -0,0 +1,110 @@
+import importlib
+import os
+import pathlib
+from typing import List
+
+from utils.common import BenchmarkConfig
+
+
+class OperatorNotFoundError(RuntimeError):
+    pass
+
+
+class BaseOperator:
+    """
+    Base class for operators.
+
+    This class defines the structure for operator implementations.
+    The forward, backward, full methods should **only contain**
+    the code that users want to benchmark.
+    """
+
+    name = None
+    variant = None
+    benchmark_config = None
+    full_name = None
+
+    def __init__(self, benchmark_config: BenchmarkConfig):
+        self.benchmark_config = benchmark_config
+        self.full_name = f"{self.name}.{self.variant}"
+
+    def forward(self):
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def backward(self):
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def full(self):
+        raise NotImplementedError("Subclasses must implement this method.")
+
+    def single_run(self):
+        """For the first input size"""
+        raise NotImplementedError("Subclasses must implement this method.")
+
+
+def dir_contains_file(dir, file_name) -> bool:
```

Comment:
It's more efficient to just check of os.path.join(dir, file_name) exists or not

Reply:
fixed in https://github.com/pytorch/pytorch/pull/136169/commits/7c2bc74a72652a45e0f4a872958d611c43ba23bd